### PR TITLE
chore: support ecommerce events in example data generator 

### PR DESCRIPTION
**Pull request by zhu-xiaowei**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary
1. Supports generating e-commerce events for the past 30 days.
2. Support real-time generation of e-commerce events.
3. Compatible with notepad event generation.

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: examples/standalone-data-generator/enums.py [hunk 1]
```diff
@@ -65,18 +62,67 @@
...
+ios_model = WeightedArray(
+    [('iPhone XR', 1.0), ('iPhone 11', 1.5), ('iPhone 11 Pro', 1.87),
+     ('iPhone 11 Pro Max', 1.3), ('iPhone 12', 2.8), ('iPhone 12 mini', 2.0),
+     ('iPhone 12 Pro', 2.8), ('iPhone 12 Pro Max', 3.2), ('iPhone 13', 4.8),
+     ('iPhone 13 mini', 2.6), ('iPhone 13 Pro', 4.2), ('iPhone 13 Pro Max', 4.2), ('iPhone 14', 6.2),
+     ('iPhone 14 Plus', 6.2), ('iPhone 14 Pro', 7.2), ('iPhone 14 Pro Max', 8.2), ('iPhone 15', 2.2),
+     ('iPhone 15 Plus', 1.2), ('iPhone 15 Pro', 3.2), ('iPhone 15 Pro Max', 2.2)])
+
+# Web browser enums
+host_name = 'd3mpqdtjegq8im.cloudfront.net'
```
Comment (zxkane) [inline examples/standalone-data-generator/enums.py:101]: use the fake one, like `example.com`

**Review by zxkane — CHANGES_REQUESTED** [review]
still see some `if` clauses in common code, like User, create_event.

Let's move the scenario code to its implementation. Extract class or interface, use inversion of control pattern to remove the dependencies.

File: examples/standalone-data-generator/configure.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+"""
+import json
+import enums
+
+APP_TYPE = enums.Application.NotePad
+
+# for history event consts
+DURATION_OF_DAYS = 30
+PER_ACTION_DURATION = range(3, 60)
+events_per_request = 10000
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply examples/standalone-data-generator/configure.py]: fixed

File: examples/standalone-data-generator/enums.py (hunk 1 above)
Comment (zhu-xiaowei) [reply examples/standalone-data-generator/enums.py]: fixed.

**Comment by zhu-xiaowei** [comment]
> still see some `if` clauses in common code, like User, create_event.
> 
> Let's move the scenario code to its implementation. Extract class or interface, use inversion of control pattern to remove the dependencies.

Added `AppProvider` and abstract behavior class `App`, and move the scenario code to  `ShoppingApp` and `NotePadApp` class which implement the interface of `App`.

File: examples/standalone-data-generator/model/App.py [hunk 3]
```diff
@@ -1,23 +1,20 @@
-"""
-Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+from abc import abstractmethod, ABC
```
Comment (zxkane) [inline examples/standalone-data-generator/model/App.py:1]: Are they from standard python library?

File: examples/standalone-data-generator/README.md [hunk 4]
```diff
@@ -1,74 +1,102 @@
...
 
-
 ### how to generate events in real-time
 
 > python3 create_event_real_time.py
 
 ### Other configuration
 
-please modify the `enums.py` to define the data size and situation
+Please modify the `configure.py` to define the data size and situation.
```
Comment (zxkane) [inline examples/standalone-data-generator/README.md:47]: situation -> scenario

File: examples/standalone-data-generator/configure.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    http://www.apache.org/licenses/LICENSE-2.0
+
+or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+and limitations under the License.
+"""
+import json
+import enums
+
+APP_TYPE = enums.Application.NotePad
```
Comment (zxkane) [inline examples/standalone-data-generator/configure.py:16]: add commented code for highlighting how switching to Shopping

**Review by zxkane — CHANGES_REQUESTED** [review]

File: examples/standalone-data-generator/model/App.py (hunk 3 above)
Comment (zhu-xiaowei) [reply examples/standalone-data-generator/model/App.py:1]: Yes, `abc` is python `Abstract Base Classes`, Reference link: https://docs.python.org/3/library/abc.html

File: examples/standalone-data-generator/configure.py (hunk 5 above)
Comment (zhu-xiaowei) [reply examples/standalone-data-generator/configure.py:16]: added.

**Review by zxkane — APPROVED** [review]

**Pull request merged**